services/nycrc/nycrc.service.js

```diff
@@ -0,0 +1,129 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { coveragePercentage } = require('../color-formatters')
+const { GithubAuthV3Service } = require('../github/github-auth-service')
+const { fetchJsonFromRepo } = require('../github/github-common-fetch')
+const { InvalidParameter } = require('..')
+
+const nycrcSchema = Joi.object({
+  branches: Joi.number()
+    .min(0)
+    .max(100),
+  lines: Joi.number()
+    .min(0)
+    .max(100),
+}).required()
+
+const pkgJSONSchema = Joi.object({
+  c8: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+  nyc: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+}).required()
+
+const documentation = `<p>
+  Create a code coverage badge, based on thresholds stored in a
+  <a href="https://github.com/istanbuljs/nyc#common-configuration-options">.nycrc config file</a>
+  on GitHub.
+</p>`
+
+module.exports = class Nycrc extends GithubAuthV3Service {
+  static get category() {
+    return 'coverage'
+  }
+
+  static get route() {
+    return {
+      base: 'nycrc',
+      // TODO: eventually add support for .yml and .yaml:
+      pattern: ':user/:repo',
+      queryParamSchema: Joi.object({
+        config: Joi.string().valid('.nycrc', '.nycrc.json', 'package.json'),
```

Comment:
👍

@paulmelnikow also suggested using a default option in his email, I think you can achieve that by chaining `default(...)` at the end if you wanted to.